modules/pytket-iqm/pytket/extensions/iqm/backends/iqm.py

```diff
@@ -0,0 +1,344 @@
+# Copyright 2020-2022 Cambridge Quantum Computing
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+from os import PathLike
+from pathlib import Path
+from typing import cast, Dict, List, Optional, Sequence, Tuple, Union
+from uuid import UUID
+from iqm_client.iqm_client import Circuit as IQMCircuit  # type: ignore
+from iqm_client.iqm_client import (  # type: ignore
+    Instruction,
+    IQMClient,
+    RunStatus,
+    SingleQubitMapping,
+)
+import numpy as np
+from pytket.backends import Backend, CircuitStatus, ResultHandle, StatusEnum
+from pytket.backends.backend import KwargTypes
+from pytket.backends.backend_exceptions import CircuitNotRunError
+from pytket.backends.backendinfo import BackendInfo
+from pytket.backends.backendresult import BackendResult
+from pytket.backends.resulthandle import _ResultIdTuple
+from pytket.circuit import Circuit, Node, OpType  # type: ignore
+from pytket.extensions.iqm._metadata import __extension_version__
+from pytket.passes import (  # type: ignore
+    BasePass,
+    SequencePass,
+    SynthesiseTket,
+    FullPeepholeOptimise,
+    FlattenRegisters,
+    RebaseCustom,
+    DecomposeBoxes,
+    RemoveRedundancies,
+    DefaultMappingPass,
+    DelayMeasures,
+    SimplifyInitial,
+)
+from pytket.predicates import (  # type: ignore
+    ConnectivityPredicate,
+    GateSetPredicate,
+    NoClassicalControlPredicate,
+    NoFastFeedforwardPredicate,
+    NoMidMeasurePredicate,
+    NoSymbolsPredicate,
+    Predicate,
+)
+from pytket.routing import Architecture  # type: ignore
+from pytket.utils import prepare_circuit
+from pytket.utils.outcomearray import OutcomeArray
+from .config import IQMConfig
+
+_GATE_SET = {OpType.PhasedX, OpType.CZ, OpType.Measure}
+
+_DEFAULT_SETTINGS = Path(__file__).resolve().parent / "demo_settings.json"
+
+# https://iqm-finland.github.io/cirq-on-iqm/api/cirq_iqm.devices.adonis.Adonis.html
+_DEFAULT_COUPLING = [
```

Comment:
Maybe add this as a comment:
      QB1
       |
QB2 - QB3 - QB4
       |
      QB5

Reply:
GitHub broke this diagram somehow, please take a look on the linked side to see the diagram I mean.

Reply:
OK.